services/spar/src/Spar/Intra/Brig.hs

```diff
@@ -96,6 +98,17 @@ toExternalId ssoid = do
   let subj = uref ^. SAML.uidSubject
   pure $ SAML.nameIDToST subj
 
+-- | Take a maybe text, a saml user ref, and construct a 'Name' from it.  If the text isn't
+-- present, use the saml subject (usually an email address).
+mkUserName :: Maybe Text -> SAML.UserRef -> Name
+mkUserName n u = fromJust $ mkUserName' n (Just u)
+
+-- | Partial variant of mkUserName.
+mkUserName' :: Maybe Text -> Maybe SAML.UserRef -> Maybe Name
+mkUserName' (Just n) _ = Just $ mkName n
```

Comment:
If this is partial anyway; why call the non-partial `mkName` that does weird things with spaces?

Reply:
Because we sometimes call 'mkName' directly, and then the totalness eliminates one more error case that everybody has to handle.

(You might argue that SCIM admins should indeed get an error if they try to set an empty user name, in which case I would argue that this is a matter of taste, and I disagree.)

Reply:
Silently setting names to ` ` sounds like a bad idea to me.  It's another weird convention we introduce. 

AFAIK the consensus is that the string `"default"` is what's used for 'empty names' across the platform. Perhaps we should use that here too?

Reply:
The only place where I see it used is in auto-provisioning saml users.   And there the old behaviuour was to set the users' name to their email address. We now break that behaviour. no? that seems bad.   This means people using SAML auto-provisioning will not see their name in the UI?

Reply:
good idea: 7530c9601206c0937ff229583c85cc399cfbe1b3

Reply:
Ah wait I can't read..  the line below sets the name to `uidSubject`

Reply:
i dont see `mkName` called anywhere else in this PR though. So I'm still confused

Reply:
8d940cc9caff77c31b52d8bb2f6d6428d90aa383